simulator/src/main/cpp/ChirpRadioPacket.h

```diff
@@ -0,0 +1,52 @@
+#define MAX_DATA_SIZE 11
+#define MAX_NEIGHBOUR_GROUPS_SIZE 17
+#define MAX_MESSAGE_SEGMENTS 128
+
+#define RADIO_ENQUIRE_PACKET 0
+#define RADIO_RESPONSE_PACKET 1
+#define RADIO_DATA_PACKET 2
+
+/**
+ * Packet Format:
+ *      Type 1 -- Enquire Packet:
+ *          type(1byte->1char->int) source(2bytes->2char->2digit int)
+ *      Type 2 -- Response Packet:
+ *          type(1byte->1char->int) source(2bytes->2char->2digit int) groupNeighbours(17bytes->array of 17 1char ints ie. 1, 2, .., 9)
+ *      Type 3 -- Data Packet:
+ *          type(1byte->1char->int) source(2bytes->2char->2digit int) groupId(1char->int) id(2bytes->2char->2digit int) sequenceNumber(2bytes->2char->2digit int) lastPacket(1byte->1char->bool) data(11bytes->11char array)
+ **/
+
+
+class ChirpRadioPacket
```

Comment:
Is this class purely getters and settters? If it is just a data structure then It may be better to just access these variables directly when required

Reply:
there are different getters and setters depending on the packet type and there are also parsing and serializing functions defined that handle different packet types.